libhb/decavcodec.c

```diff
@@ -1426,6 +1433,61 @@ static int decodeFrame( hb_work_private_t * pv, packet_info_t * packet_info )
     return got_picture;
 }
 
+#if HB_PROJECT_FEATURE_NVENC
+static enum AVPixelFormat get_hw_pix_fmt(AVCodecContext *ctx,
+                                         const enum AVPixelFormat *pix_fmts) {
+  const enum AVPixelFormat *p;
+
+  for (p = pix_fmts; *p != -1; p++) {
+    if (*p == AV_PIX_FMT_CUDA) {
+      return *p;
+    }
+  }
+
+  hb_error("Failed to get HW surface format.\n");
+  return AV_PIX_FMT_NONE;
+}
+
+static int cuda_hw_ctx_init(AVCodecContext *ctx, AVBufferRef *hw_device_ctx,
+                            const enum AVHWDeviceType type) {
+  int err = av_hwdevice_ctx_create(&hw_device_ctx, type, NULL, NULL, 0);
+  if (err < 0) {
+    hb_error("Failed to create specified HW device.\n");
+    return err;
+  }
+
+  ctx->hw_device_ctx = av_buffer_ref(hw_device_ctx);
+  return err;
+}
+
+static const char *find_cuvid_codec_by_name(enum AVCodecID id) {
+  switch (id) {
+  case AV_CODEC_ID_AV1:
+    return "av1_cuvid";
```

Comment:
What would happen if we run this code on say a GTX1080 that doesn't have AV1 decoder? Will it fallback to software decoding fine?

Reply:
libavcodec will return an error while opening the codec:

```bash
[av1_cuvid @ 0x7fffe4003dc0] Codec av1_cuvid is not supported.
```

>Will it fallback to software decoding fine?

It's not implemented yet but I understand that fallback behavior shall be added.